packages/core/saved-objects/core-saved-objects-server/src/spaces.ts

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+/**
+ * The ISavedObjectsSpacesExtension interface defines the functions of a saved objects repository spaces extension.
+ * It contains functions for getting the current namespace & getting and array of searchable spaces.
+ */
+export interface ISavedObjectsSpacesExtension {
+  /**
+   * Retrieves the active namespace ID. This is *not* the same as a namespace string. See also: `namespaceIdToString` and
+   * `namespaceStringToId`.
+   *
+   * This takes the saved objects repository's namespace option as a parameter, and doubles as a validation function; if the namespace
+   * option has already been set some other way, this will throw an error.
+   */
+  getCurrentNamespace: (namespace: string | undefined) => string | undefined;
```

Comment:
I'm not sure I follow. In this case we don't want a setter. This function will return the active space if you pass in undefined, otherwise will throw an error. It's a bit a weird function, but serves the niche purpose of making sure the namespace parameter for repository calls is not being used if the spaces feature is active. The active space is only set via the constructor.

Perhaps the comment just needs to be updated?

Reply:
> It's a bit a weird function, but serves the niche purpose of making sure the namespace parameter for repository calls is not being used if the spaces feature is active

This is what I was referring to. This strange get/throw-if-set pattern was made because of the limitations of the SO wrappers at the time. Given wrappers are no more, we could eventually switch to a more well defined / less hacky interface for the spaces extension.

But it would be fine keeping it unchanged